### PR TITLE
[Dev] Move function from testUtils.ts to gameManager.ts (1/3)

### PR DESCRIPTION
**Pull request by torranx**
## What are the changes?
- created base class for game helpers
- created move helper class

## Why am I doing these changes?
- I feel like testUtils should be used for general utils and not game/battle-specific. 
- The function is directly related to phases which the GameManager class describes `manage the game state and transitions between phases`
- the function could then be directly accessible to the `game` instance instead of manually importing it ie `game.move.forceHit()`

This removes:
- phases imports
- parameter `game` from the function

## What did change?
see above

### Screenshots/Videos

n/a

## How to test the changes?

`$ npm run test`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/test/utils/gameManager.ts [hunk 1]
```diff
@@ -354,4 +355,19 @@ export default class GameManager {
...
+   * succeeding `MoveEffectPhase` immediately following the first one
+   * (in the case of a multi-target move)
+   *
+   * @param shouldHit Whether the move should hit
+   */
+  async mockHitCheck(shouldHit: boolean): Promise<void> {
+    await this.phaseInterceptor.to(MoveEffectPhase, false);
+
+    vi.spyOn(this.scene.getCurrentPhase() as MoveEffectPhase, "hitCheck").mockReturnValue(shouldHit);
+  }
```
Comment (flx-sta) [inline src/test/utils/gameManager.ts:372]: I think the naming of the method is a little bit unfortunate combined with the param name.

How about  `nextMoveShouldHit(shouldHit: boolean)`

Or does it mock all hit checks?

then `movesShouldHit(shouldHit: boolean)`

----

### Even better would be to call it like this:

```ts
game.moves.guaranteeHit(); 
game.moves.guaranteeMiss(); 
```
or
```ts
game.move.guaranteeHit(); 
game.move.guaranteeMiss(); 
```

Depending on if plural or singular mock

Comment (torranx) [reply src/test/utils/gameManager.ts:372]: I'll try and implement `game.move.guaranteeHit()` but I feel lie the word guarantee is quite long, maybe `game.move.shouldHit()`? Also I'm yet to find a way to mock _all_ moves, if you have an idea to share that'd be great.

Comment (torranx) [reply src/test/utils/gameManager.ts:372]: Actually, I could just make a new override

Comment (flx-sta) [reply src/test/utils/gameManager.ts:372]: Unfortunately I know as much as you do :D
If you are using an override put it into overridehelper

**Review by flx-sta — CHANGES_REQUESTED** [review]
@torranx pelase check `make_it_rain.test.ts`

```ts
  it("should reduce Sp. Atk if it only hits the second target", async () => {
    await game.startBattle([Species.CHARIZARD, Species.BLASTOISE]);

    const playerPokemon = game.scene.getPlayerField();

    game.doAttack(getMovePosition(game.scene, 0, Moves.MAKE_IT_RAIN));
    game.doAttack(getMovePosition(game.scene, 1, Moves.SPLASH));

    await game.phaseInterceptor.to(MoveEffectPhase, false);

    // Make Make It Rain miss the first target
    const moveEffectPhase = game.scene.getCurrentPhase() as MoveEffectPhase;
    vi.spyOn(moveEffectPhase, "hitCheck").mockReturnValueOnce(false); // ⬅️⬅️⬅️⬅️⬅️

    await game.phaseInterceptor.to(MoveEndPhase);

    expect(playerPokemon[0].summonData.battleStats[BattleStat.SPATK]).toBe(-1);
  }, TIMEOUT);
  ```

**Comment by torranx** [comment]
> @torranx pelase check `make_it_rain.test.ts`
> 
> ```ts
>   it("should reduce Sp. Atk if it only hits the second target", async () => {
>     await game.startBattle([Species.CHARIZARD, Species.BLASTOISE]);
> 
>     const playerPokemon = game.scene.getPlayerField();
> 
>     game.doAttack(getMovePosition(game.scene, 0, Moves.MAKE_IT_RAIN));
>     game.doAttack(getMovePosition(game.scene, 1, Moves.SPLASH));
> 
>     await game.phaseInterceptor.to(MoveEffectPhase, false);
> 
>     // Make Make It Rain miss the first target
>     const moveEffectPhase = game.scene.getCurrentPhase() as MoveEffectPhase;
>     vi.spyOn(moveEffectPhase, "hitCheck").mockReturnValueOnce(false); // ⬅️⬅️⬅️⬅️⬅️
> 
>     await game.phaseInterceptor.to(MoveEndPhase);
> 
>     expect(playerPokemon[0].summonData.battleStats[BattleStat.SPATK]).toBe(-1);
>   }, TIMEOUT);
> ```



> @torranx pelase check `make_it_rain.test.ts`
> 
> ```ts
>   it("should reduce Sp. Atk if it only hits the second target", async () => {
>     await game.startBattle([Species.CHARIZARD, Species.BLASTOISE]);
> 
>     const playerPokemon = game.scene.getPlayerField();
> 
>     game.doAttack(getMovePosition(game.scene, 0, Moves.MAKE_IT_RAIN));
>     game.doAttack(getMovePosition(game.scene, 1, Moves.SPLASH));
> 
>     await game.phaseInterceptor.to(MoveEffectPhase, false);
> 
>     // Make Make It Rain miss the first target
>     const moveEffectPhase = game.scene.getCurrentPhase() as MoveEffectPhase;
>     vi.spyOn(moveEffectPhase, "hitCheck").mockReturnValueOnce(false); // ⚠️⚠️⚠️
> 
>     await game.phaseInterceptor.to(MoveEndPhase);
> 
>     expect(playerPokemon[0].summonData.battleStats[BattleStat.SPATK]).toBe(-1);
>   }, TIMEOUT);
> ```

updated forceMiss() for this

**Review by flx-sta — APPROVED** [review]
lgtm 🚀

**Pull request merged**